docs/site/HasMany-relation.md

```diff
@@ -0,0 +1,206 @@
+> {
+  public orders: HasManyRepositoryFactory<Order, typeof Customer.prototype.id>;
+  constructor(
+    @inject('datasources.db') protected db: juggler.DataSource,
+    @repository(OrderRepository) orderRepository: OrderRepository,
+  ) {
+    super(Customer, db);
+    this.orders = this._createHasManyRepositoryFactoryFor(
+      'orders',
+      orderRepository,
+    );
+  }
+}
+```
+
+The following CRUD APIs are now available in the constrained target repository
+factory `orders` for instances of `customerRepository`:
+
+- `create` for creating a target model instance belonging to customer model
+  instance
+  ([API Docs](https://apidocs.strongloop.com/@loopback%2fdocs/repository.html#DefaultHasManyEntityCrudRepository.prototype.create))
+- `find` finding target model instance(s) belonging to customer model instance
+  ([API Docs](https://apidocs.strongloop.com/@loopback%2fdocs/repository.html#DefaultHasManyEntityCrudRepository.prototype.find))
+- `delete` for deleting target model instance(s) belonging to customer model
+  instance
+  ([API Docs](https://apidocs.strongloop.com/@loopback%2fdocs/repository.html#DefaultHasManyEntityCrudRepository.prototype.delete))
+- `patch` for patching target model instance(s) belonging to customer model
+  instance
+  ([API Docs](https://apidocs.strongloop.com/@loopback%2fdocs/repository.html#DefaultHasManyEntityCrudRepository.prototype.patch))
+
+## Using hasMany constrained repository in a controller
+
+The same pattern used for ordinary repositories to expose their CRUD APIs via
+controller methods is employed for `hasMany` repositories. Once the hasMany
+relation has been defined and configured, controller methods can call the
+underlying constrained repository CRUD APIs and expose them as routes once
+decorated with
+[Route decorators](Routes.md#using-route-decorators-with-controller-methods). It
+will require the value of the foreign key and, depending on the request method,
+a value for the target model instance as demonstrated below.
+
+{% include code-caption.html
+content="src/controllers/customer-orders.controller.ts" %}
+
+```ts
+import {post, param, requestBody} from '@loopback/rest';
+import {customerRepository} from '../repositories/customer.repository.ts';
+import {Customer, Order} from '../models/';
+
+export class CustomerOrdersController {
+  constructor(
+    @repository(CustomerRepository)
+    protected customerRepository: CustomerRepository,
+  ) {}
+
+  @post('/customers/{id}/order')
+  async createOrder(
+    @param.path.number('id') customerId: typeof Customer.prototype.id,
+    @requestBody() orderData: Order,
+  ): Promise<Order> {
+    return await this.customerRepository.orders(customerId).create(orderData);
+  }
+}
+```
+
+In LoopBack 3, the REST APIs for relations were exposed using static methods
+with the name following the pattern `__{methodName}__{relationName}__`, for
+example `Customer.__find__orders`. We recommend to create a new controller for
```

Comment:
Line break here please. The latter paragraph might be better as a note, but not a big deal

Reply:
I don't think a line break is needed here since the first sentence is supposed to be an opener for the whole paragraph.